src/main/groovy/cruise/umple/UmpleGradlePlugin.groovy

```diff
@@ -29,12 +29,14 @@ class UmpleGradlePlugin implements Plugin<Project> {
 		project.task('generateSource') << {		
 			// command line arguments can be specified through gradle by -P (-P is for project properties)
 			// eg: "gradle compileUmpleFile -PUMPLE_FILE_PATH=test.ump -PLANGUAGE_TO_GENERATE=Java"	
+			m_umpleFilePath = "${project.projectDir}${File.separator}" 
+			m_generatedOutputPath = "${project.projectDir}${File.separator}"
 		
 			if(project.hasProperty(UMPLE_FILE_PATH))
 			{
-				m_umpleFilePath = project.getProperty(UMPLE_FILE_PATH)	
+				m_umpleFilePath += project.getProperty(UMPLE_FILE_PATH)	
```

Comment:
I'm not sure what you're getting at. The only way I've been able to avoid problems is by making the path absolute. Ideally, I'd declare everything on a single line before the plugin's `apply` method, but because I don't have access to the `project.projectDir` until I'm inside the plugin's `apply` method I need to do it this way.

Are you just suggesting I replace the string concatenation with calls to `relativize`?